awx/main/models/ha.py

```diff
@@ -509,10 +526,12 @@ def on_instance_saved(sender, instance, created=False, raw=False, **kwargs):
     2. a node changes its value of peers_from_control_nodes
     3. a new control node comes online and has instances to peer to
     '''
+    from awx.main.signals import disable_activity_stream
+
     if created and settings.IS_K8S and instance.node_type in [Instance.Types.CONTROL, Instance.Types.HYBRID]:
-        inst = Instance.objects.filter(peers_from_control_nodes=True)
-        if set(instance.peers.all()) != set(inst):
-            instance.peers.set(inst)
+        peers_address = ReceptorAddress.objects.filter(peers_from_control_nodes=True)
+        with disable_activity_stream():
+            instance.peers.add(*peers_address)
```

Comment:
The uniqueconstraint on `<source, target>` ensures that the list of `ReceptorAddress` related to the instance is, effectively, a set. :+1: 

My worry was that this might add duplicates, but I've convinced myself that is not the case. :+1:

Reply:
yeah the unique constraint is necessary here to prevent duplicate peering